python_modules/dagster/dagster/_core/storage/partition_status_cache.py

```diff
@@ -197,6 +219,80 @@ def _build_status_cache(
         latest_storage_id=latest_storage_id,
         partitions_def_id=partitions_def.serializable_unique_identifier,
         serialized_materialized_partition_subset=serialized_materialized_partition_subset.serialize(),
+        serialized_failed_partition_subset=_get_failed_partitions(
+            instance, asset_key, partitions_def
+        ).serialize(),
+    )
+
+
+def _filter_incomplete_materialized_runs_to_failed(
+    instance: DagsterInstance, incomplete_materialization_runs: Mapping[str, str]
+) -> Set[str]:
+    failed_run_ids = {
+        r.run_id
+        for r in instance.get_runs(
+            filters=RunsFilter(
+                run_ids=list(incomplete_materialization_runs.values()),
+                statuses=[DagsterRunStatus.FAILURE],
+            )
+        )
+    }
+
+    return {p for p, run_id in incomplete_materialization_runs.items() if run_id in failed_run_ids}
+
+
+def _get_failed_partitions(
+    instance: DagsterInstance, asset_key: AssetKey, partitions_def: PartitionsDefinition
+) -> PartitionsSubset:
+    incomplete_materialization_runs = instance.event_log_storage.get_latest_asset_partition_materialization_attempts_without_materializations(
+        asset_key
+    )
+
+    if not incomplete_materialization_runs:
+        return partitions_def.empty_subset()
+
+    failed_partitions = _filter_incomplete_materialized_runs_to_failed(
+        instance, incomplete_materialization_runs
+    )
+
+    return partitions_def.empty_subset().with_partition_keys(
+        get_validated_partition_keys(instance, partitions_def, failed_partitions)
+    )
+
+
+def _get_updated_failed_partition_subset(
+    instance: DagsterInstance,
+    asset_key: AssetKey,
+    partitions_def: PartitionsDefinition,
+    current_cached_subset: PartitionsSubset,
+    unevaluated_planned_event_records: Sequence[EventLogRecord],
```

Comment:
can we rename this arg to unevaluated_event_records? since it contains both materialization and materialization planned events